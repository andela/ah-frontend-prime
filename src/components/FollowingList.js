import React from "react";

const FollowingList = props => {
  const { following } = props;
  return (
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title">My Following</h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          {following && following.length > 0
            ? following.map(item => (
                <div
                  className="row"
                  id="following"
                  key={item.username}
                  onClick={() => {
                    document.location.href = `/profile/${item.username}`;
                  }}
                  key={item.username}
                >
                  <button className="col col-lg-3" style={{ border: "none" }}>
                    <img
                      src={
                        item.image
                          ? `${item.image}`
                          : "https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584__340.png"
                      }
                      alt="profile"
                      className="profile-page"
                    />
                  </button>
                  <div className="col col-lg-9">
                    <h5>{item.username}</h5>
                    <p>{item.full_name}</p>
                  </div>
                  <br />
                </div>
              ))
            : "No one is on the following List"}
        </div>
        <div class="modal-footer" />
      </div>
    </div>
  );
};

export default FollowingList;
