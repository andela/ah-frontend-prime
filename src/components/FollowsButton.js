import React from "react";

const FollowsButton = props => {
  const { isfollowing, unfollowUser, followUser } = props;
  return (
    <button
      onClick={
        isfollowing
          ? unfollowUser.bind(this, sessionStorage.getItem("userview_name"))
          : followUser.bind(this, sessionStorage.getItem("userview_name"))
      }
      className="btn btn-outline-success"
    >
      {isfollowing ? "Unfollow" : "Follow"}
    </button>
  );
};

export default FollowsButton;
