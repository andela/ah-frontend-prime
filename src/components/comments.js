import React, { Component } from "react";
import moment from "moment";
import { articleComments, fetchComments } from "../actions/commentsActions";
import { connect } from "react-redux";
import "../styles/comments.scss";


export class Comments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comments: ""
    };
  }

  componentDidMount() {
    const { fetchComments, articleSlug } = this.props;
    fetchComments(articleSlug);
  }

  handleOnChange = event => {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleOnSubmit = event => {
    event.preventDefault();
    const { articleSlug, articleComments, fetchComments } = this.props;
    const { comments } = this.state;
    const data = {
      body: comments
    };
    articleComments(data, articleSlug);
    fetchComments(articleSlug);
  };

  render() {
    const { commentsMessage } = this.props;
    const comment = commentsMessage.map(message => (
      <section className="comments" key={message.id}>
        <article className="comment">
          <a className="comment-img" href="#non">
            <img src={message.author.image} alt="" width="50" height="50" />
          </a>
          <div className="comment-body">
            <div className="text">
              <p>{message.body}</p>
            </div>
            <p className="attribution">
              by <a href={`/profile/${message.author.username}`}>{message.author.username}</a> at {moment(message.createdAt).format(
                          "h:mm:ss a [on] MMMM Do YYYY.")}
                        
            </p>
          </div>
        </article>
      </section>
    ));
    return (
      <div className="row">
        <div className="col col-lg-1" />
        <div className="col col-lg-10">
          <div className="row">
            <div className="col col-lg-1" />
            <div className="col col-lg-11">
              <form
                method="POST"
                onSubmit={this.handleOnSubmit}
                className="comment-forms"
              >
                <textarea
                  className="form-control"
                  name="comments"
                  value={this.state.comments}
                  onChange={this.handleOnChange}
                >
                  comments here
                </textarea>
                <button
                  type="submit"
                  className="btn btn-comment"
                  style={{ marginTop: "10px" }}
                >
                  comment
                </button>
              </form>
            </div>
          </div>
          {comment}
        </div>
        <div className="col col-lg-1" />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  commentsSuccess: state.comments.success,
  commentsErrors: state.comments.errors,
  commentsMessage: state.getComments.comments
});
export default connect(
  mapStateToProps,
  { articleComments, fetchComments }
)(Comments);
