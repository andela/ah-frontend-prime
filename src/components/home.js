import React, { Component } from "react";
import { connect } from "react-redux";
import "../styles/app.scss";
import moment from "moment";
import { Link } from "react-router-dom";
import { getArticlesAction } from "../actions/getArticles";

export class Home extends Component {
  constructor(props) {
    super(props);
    this.fetchNext = this.fetchNext.bind(this);
    this.fetchPrevious = this.fetchPrevious.bind(this);
  }
  componentDidMount() {
    const { getArticlesAction } = this.props;
    getArticlesAction();
  }
  fetchNext() {
    const { next, getArticlesAction } = this.props;
    getArticlesAction(next);
  }
  fetchPrevious() {
    const { previous, getArticlesAction } = this.props;
    getArticlesAction(previous);
  }
  render() {
    const { articles } = this.props;
    const articleList = articles.length ? (
      articles.map(article => {
        return (
          <div className="article" key={article.slug}>
            <div className="inner">
              <Link
                className="links"
                to={{
                  pathname: "/article/" + article.slug
                }}
              >
                <div className="article-card">
                  <h3>{article.title.substring(0, 30)} ...</h3>
                  <p>
                    <b>{article.description.substring(0, 60)} ...</b>
                  </p>
                  <span>
                    <b>Author: {article.author.username}</b>
                  </span>
                  <br />

                  <span className="card-date">
                    Created at{" "}
                    {moment(article.createdAt).format(
                      "h:mm:ss a [on] MMMM Do YYYY."
                    )}
                  </span>
                  <br />
                  <p>
                    <b>{article.reading_time}</b>
                  </p>
                </div>

                <div>
                  <img
                    src={
                      article.image
                        ? article.image
                        : "https://previews.123rf.com/images/twindesigner/twindesigner1708/twin" +
                          "designer170800135/84202763-ah-brush-letter-logo-design-with-black-" +
                          "circle-creative-brushed-letters-icon-logo-.jpg"
                    }
                    className="img-tile"
                  />
                </div>
              </Link>
            </div>
          </div>
        );
      })
    ) : (
      <div>No Articles in Authors Haven Yet</div>
    );

    return (
      <div className="big-container">
        <div className="articles-top">
          <div className="best-article">
            <div className="inner">
              <img
                src={
                  "http://allpicts.in/download/22916/2018/03/Natural_Images_HD_1" +
                  "080p_Download_with_Waterfall_in_Tollymore_Forest_Park-1440x900.jpg/"
                }
                className="img-top"
              />
              <div className="inner">
                <div className="inner">
                  <div className="best-article-card">
                    <h3>
                      Facebook's Plan to Fuse its Messaging Apps Is Not about
                      Your Privacy Just follow the Money
                    </h3>
                    <span>
                      <b>Author: Yasha Levin in OneZero</b>
                    </span>
                    <br />
                    <span className="card-date">Created a few hours ago</span>
                  </div>
                </div>
              </div>
            </div>
            <div />
          </div>
          <div className="top-articles">{articleList}</div>
          <div className="top-articles">{articleList}</div>
        </div>

        <div className="articles">
          <div className="heading">
            <h3>Featured Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div className="articles1">
          <div className="heading">
            <h3>Popular Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div>
          <button
            className="btn btn-outline-success btn eian"
            onClick={() => this.fetchPrevious()}
          >
            Previous
          </button>{" "}
          <button
            className="btn btn-outline-success btn aken"
            onClick={() => this.fetchNext()}
          >
            Next
          </button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.getArticlesReducer.articles,
  next: state.getArticlesReducer.next,
  previous: state.getArticlesReducer.previous
});

export default connect(
  mapStateToProps,
  { getArticlesAction }
)(Home);
