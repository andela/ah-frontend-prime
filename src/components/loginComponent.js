import React from 'react';
import '../styles/app.scss';

const Login = () => {
  return (
    <div>
      <h2>Login</h2>
    </div>
	)
}

export default Login;
