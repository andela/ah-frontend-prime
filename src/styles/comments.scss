article, aside, figure, footer, header, hgroup, menu, nav, section {
  display: block
}

a {
  color: #6d84b4;
  text-decoration: none;
}

section {
  background: rgba(255, 255, 255, 0.9);
  padding-right: 85px;
  padding-top: 20px;
  padding-bottom: 20px;
  height: auto;
  width: 100%;
  margin: auto;
  margin-top: 10px;
  border-radius: 5px;
}

.comment {
  overflow: hidden;
  padding: 0 0 1em;
  border-bottom: 1px solid #ddd;
  margin: 0 0 1em 40px;
  *zoom: 1;
  width: 100%;
}

.comment-img {
  float: left;
  margin-right: 33px;
  border-radius: 50%;
  border: 2px solid #e5e5e5;
  overflow: hidden;
}

.comment-img img {
  display: block;
}

.comment-body {
  overflow: hidden
}

.comment .text {
  padding: 10px;
  border: 1px solid #e5e5e5;
  border-radius: 5px;
  background: #fff;
}

.comment .text p:last-child {
  margin: 0
}

.comment .attribution {
  margin: 0.5em 0 0;
  font-size: 14px;
  color: #666;
}

/* Decoration */

.comments, .comment {
  position: relative
}

.comments:before, .comment:before, .comment .text:before {
  content: "";
  position: absolute;
  top: 0;
  left: 65px;
}

.comments:before {
  width: 3px;
  left: 105px;
  bottom: 0px;
  background: rgba(0, 0, 0, 0.1);
}

.comment:before {
  width: 9px;
  height: 9px;
  border: 3px solid #fff;
  border-radius: 100px;
  margin: 16px 0 0 -6px;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.2), inset 0 1px 1px rgba(0, 0, 0, 0.1);
  background: #ccc;
}

.comment:hover {
  cursor: cell;
}

.comment:hover:before {
  background: #3b5998
}

.comment .text:before {
  top: 18px;
  left: 78px;
  width: 9px;
  height: 9px;
  border-width: 0 0 1px 1px;
  border-style: solid;
  border-color: #e5e5e5;
  background: #fff;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -o-transform: rotate(45deg);
}

.comment-forms {
  textarea.form-control {
    border: 1px solid #e5e5e5;
    height: 100px;
  }

  .btn-comment {
    border: 1px solid #ccc;
    float: right;
  }
}
