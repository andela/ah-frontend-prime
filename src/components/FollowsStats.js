import React from "react";
import FollowingList from "./FollowingList";
import FollowersList from "./FollowersList";

const FollowsStats = props => {
  const { following_count, followers_count, followers, following } = props;
  return (
    <div className="row">
      <div
        className="modal fade"
        id="exampleModal"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowingList following={following} />
      </div>
      <div
        className="modal fade"
        id="exampleModal1"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowersList followers={followers} />
      </div>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal1"
        href="#"
      >
        {followers_count === 1 ? "Follower" : "Followers"}
        {" : "}
        <button className="stats">
          <strong>{followers_count ? followers_count : 0} </strong>
        </button>
      </a>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal"
        href="#"
      >
        Following{" : "}
        <button className="stats">
          <strong>{following_count ? following_count : "0"} </strong>
        </button>
      </a>
    </div>
  );
};

export default FollowsStats;
