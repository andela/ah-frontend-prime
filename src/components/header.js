import React from "react";
import PropTypes from "prop-types";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import { fade } from "@material-ui/core/styles/colorManipulator";
import { withStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";

const styles = theme => ({
  root: {
    width: "100%"
  },
  grow: {
    flexGrow: 1
  },
  menu: {
    marginLeft: -12,
    marginRight: 20
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block"
    }
  },
  inputRoot: {
    color: "inherit",
    width: "100%"
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 9,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: 118,
      "&:focus": {
        width: 180
      }
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 8,
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  bar: {
    backgroundColor: "#003153",
    boxShadow: "none"
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.18),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.29)
    },
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing.unit,
      width: "auto"
    }
  }
});

function TopAppBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static" className={classes.bar}>
        <Toolbar>
          <Typography
            className={classes.title}
            variant="h6"
            color="inherit"
            noWrap
          >
            Authors Haven
          </Typography>
          <div className={classes.grow} />

          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput
              }}
            />
          </div>
          <div className="nav-div">
            <a href="login" className="nav-links">
              Sign In
            </a>
            &nbsp;&nbsp; | &nbsp;&nbsp;
            <a href="register" className="nav-links">
              Get Started
            </a>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
}

TopAppBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(TopAppBar);
