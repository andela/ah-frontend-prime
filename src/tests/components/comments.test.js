import React from "react";
import { shallow } from "enzyme";
import { Comments, mapStateToProps } from "../../components/comments";
import { commentData, fetchAllComments } from "../mock_data/moxios_mock";

describe("test for comments", () => {
  let wrapper;
  let instance;
  let props = {
    comments: "",
    fetchComments: jest.fn(),
    handleOnChange: jest.fn(),
    articleComments: jest.fn(),
    commentsMessage: [],
    mapStateToProps: jest.fn()
  };
  beforeEach(() => {
    wrapper = shallow(<Comments {...props} />);
    instance = wrapper.instance();
  });
  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should test the handleChange method", () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: "comments",
        value: "great article"
      }
    };
    instance.handleOnChange(event);
    expect(instance.state.comments).toEqual("great article");
  });
  it("should test the onSubmit method ", () => {
    const event = {
      preventDefault: jest.fn()
    };
    instance.handleOnSubmit(event);
    expect(props.articleComments).toHaveBeenCalled();
  });
  it("should ", () => {
    wrapper.setProps({
      commentsMessage: [
        {
          id: 2,
          body: "this is it",
          author: {
            image: ""
          }
        }
      ]
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("should test map state to props", () => {
    let state;
    state = {
      getComments: {
        comments: fetchAllComments
      },
      comments: {
        success: commentData
      }
    };
    mapStateToProps(state);
    expect(props.mapStateToProps).toHaveBeenCalledTimes(0);
  });
});
