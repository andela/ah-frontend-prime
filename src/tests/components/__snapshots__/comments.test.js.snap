// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for comments should  1`] = `ShallowWrapper {}`;

exports[`test for comments should render without crashing 1`] = `ShallowWrapper {}`;
